test/validation/api/ipsec/ipsec.h

```diff
@@ -41,7 +41,7 @@ typedef struct {
 	uint32_t l2_offset;
 	uint32_t l3_offset;
 	uint32_t l4_offset;
-	uint8_t data[256];
+	uint8_t data[384];
```

Comment:
@JannePeltonen This field was updated to handle packet size requirement for IPv6 + UDP encap.

For reassembly UT we need to have larger packet sizes and so we need to either update this to large value (like 5800 for example) or replicate this struct as a ipsec_reass_test_packet. Replicating the struct would mean lot of functions also would get replicated. So I would like to update this field for that case.

This change would increase only the standard vectors size. Buffer size used for packet pool creation would be unaffected for the existing UTs.

Can you confirm if updating this field is fine.

Reply:
I do not see a problem in increasing the size of the array.

However, it is hard to verify that 384 or any other value is enough. The compiler will complain if the initializers in the test vectors exceed the size, but ipsec_check_out_in_one() writes to the array without doing any length check. I would suggest defining the array size through a macro and checking packet lenght against it in ipsec_check_out_in_one() before the odp_packet_copy_to_mem() call.

Reply:
It makes sense to introduce macros to indicate the max size of the array. 

> The compiler will complain if the initializers in the test vectors exceed the size, but ipsec_check_out_in_one() writes
> to the array without doing any length check.

Would this be sufficient? 
https://github.com/OpenDataPlane/odp/blob/master/test/validation/api/ipsec/ipsec.c#L898

The sizeof(pkt_in.data) would be same as the size of the array, right?

Reply:
Yes, that is sufficient, so you can ignore my comment about there not being a check. I just missed it because it is not close to the copy operation for which it is needed.